frontend/src/layouts/page-layout/PageLayout.jsx

```diff
@@ -1,34 +1,159 @@
+      headers: {
+        "X-CSRFToken": sessionDetails.csrfToken,
+      },
+    };
+
+    try {
+      const response = await axios(requestOptions);
+      return response.data;
+    } catch (error) {
+      return { data: [] };
+    }
+  };
+
+  const closeLogsModal = () => {
+    setShowLogsModal(false);
+  };
+  const handleCollapse = (keys) => {
+    setActiveKey(keys);
+    setHeight(keys.length > 0 ? 200 : 50);
+  };
+
+  const getItems = () => [
+    {
+      key: "1",
+      label:
+        activeKey?.length > 0 ? (
+          <>
+            <div
+              aria-hidden="true"
+              style={{
+                position: "absolute",
+                width: "100%",
+                top: 0,
+                cursor: "row-resize",
+                height: "10px",
+              }}
+              onMouseDown={enableResize}
+              onClick={(e) => e.stopPropagation()}
+            />
+            <LogsLabel />
+          </>
+        ) : (
+          <Typography className="logs-title">
+            <ExceptionOutlined />
+            Logs
+          </Typography>
+        ),
+      children: (
+        <div className="agency-ide-logs" style={{ height: height - 50 }}>
+          <DisplayLogs />
+        </div>
+      ),
+    },
+  ];
+
   useEffect(() => {
     localStorage.setItem("collapsed", JSON.stringify(collapsed));
-  }, [collapsed]);
+    if (activeKey.length > 0) {
+      getLogs().then((res) =>
+        useSocketLogsStore.setState(() => ({
+          logs: [...res.data],
+        }))
+      );
+    }
+  }, [collapsed, activeKey]);
 
   return (
     <div className="landingPage">
       <TopNavBar />
       <Layout>
         <SideNavBar collapsed={collapsed} />
-        <Layout>
+        <Layout className="overflow-hidden">
           <Button
             shape="circle"
             size="small"
             icon={collapsed ? <RightOutlined /> : <LeftOutlined />}
             onClick={() => setCollapsed(!collapsed)}
             className="collapse_btn"
           />
-          <Outlet />
+          <div className="page-layout-body overflow-hidden">
+            <div className="page-layout-main overflow-hidden">
+              <Outlet />
+            </div>
+            <Footer className="log-footer">
+              <Collapse
+                className={`ide-collapse-panel ${
+                  isBlink ? "blinking-border" : ""
+                }`}
+                size="small"
+                activeKey={activeKey}
+                items={getItems()}
+                expandIconPosition="end"
+                onChange={handleCollapse}
+                bordered={false}
+                style={{ height: height }}
+              />
+              <Modal
+                title="Logs"
+                open={showLogsModal}
+                onCancel={closeLogsModal}
+                className="agency-ide-log-modal"
+                footer={null}
+                width={1000}
```

Comment:
Why width 1000 ? @jagadeeswaran-zipstack

Reply:
@vishnuszipstack That is the width for logs modal. @tahierhussain would be right person to address here.

Reply:
@vishnuszipstack This is the width we've set for the logs component. It will be responsive, meaning if the screen shrinks, the width will be automatically adjusted in Antd's Modal component.